catalog/app/components/JsonEditor/State.js

```diff
@@ -63,13 +60,13 @@ function mapKeys(objectOrArray, callback, schemaKeys) {
     (memo, key) => (memo.includes(key) ? memo : memo.concat(key)),
     Object.keys(objectOrArray),
   )
-  return keys
-    .sort((a, b) => getSortIndex(a) - getSortIndex(b))
-    .map((key) => callback(objectOrArray[key], key, schemaSort[key]))
+  const sortFunc = R.sort((a, b) => getSortIndex(a) - getSortIndex(b))
+  const mapFunc = R.map((key) => callback(objectOrArray[key], key, schemaSort[key]))
+  return R.compose(sortFunc, mapFunc)(keys)
```

Comment:
fyi there's a convenience helper in `utils/pipeThru` that helps to express such cases in a slightly more readable and intuitive way (imo), like this:
```javascript
return pipeThru(keys)(
  R.map((key) => callback(objectOrArray[key], key, schemaSort[key])),
  R.sortBy(getSortIndex),
) 
```

ofc pipeline operator would make this even better